go/nomdl/lexer.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package nomdl
+
+import (
+	"fmt"
+	"text/scanner"
+)
+
+type token struct {
+	Pos  scanner.Position
+	Type rune
+	Text string
+}
+
+func (t token) String() string {
+	return t.Text
+}
+
+type lexer struct {
+	scanner   *scanner.Scanner
+	nextToken token
+}
+
+func new(s *scanner.Scanner) *lexer {
```

Comment:
This package has `new()` and `New()`, which seems suboptimal. Also...isn't `new` a reserved word in Go?